spi/topology/src/main/java/quarks/topology/spi/graph/TWindowImpl.java

```diff
@@ -30,4 +35,20 @@
         Aggregate<T,U,K> op = new Aggregate<T,U,K>(window, processor);
         return feeder().pipe(op); 
     }
+
+    @Override
+    public <U> TStream<U> batch(BiFunction<List<T>, K, U> batcher) {
+        batcher = Functions.synchronizedBiFunction(batcher);
+        Window<T, K, List<T>> window =
+                Windows.window(
```

Comment:
Right. The runtime invokes the contents policy before adding the tuple:
``` Java
getWindow().getContentsPolicy().accept(this, tuple);
this.tuples.add(tuple);
// Trigger
getWindow().getTriggerPolicy().accept(this, tuple);
```
If there are 9 tuples in a window when window.insert() is called, the contents policy will see no need to schedule an evict. The tuple will then be inserted, the partition processed, and its contents evicted the next time a tuple is inserted.

For a time-based window:
the contents policy kicks off a recurring eviction every `time` seconds. Each eviction does three things:
* Calls partition.process()
* Evicts all tuples
* Schedules itself to be called `time` seconds from now.